test/extensions/filters/http/dynamic_forward_proxy/proxy_filter_integration_test.cc

```diff
@@ -225,43 +266,22 @@ INSTANTIATE_TEST_SUITE_P(IpVersions, ProxyFilterIntegrationTest,
 
 // A basic test where we pause a request to lookup localhost, and then do another request which
 // should hit the TLS cache.
-TEST_P(ProxyFilterIntegrationTest, RequestWithBody) {
-  int64_t original_usec = dispatcher_->timeSource().monotonicTime().time_since_epoch().count();
-
-  config_helper_.prependFilter(fmt::format(R"EOF(
-  name: stream-info-to-headers-filter
-)EOF"));
-
-  initializeWithArgs();
-  codec_client_ = makeHttpConnection(lookupPort("http"));
-  const Http::TestRequestHeaderMapImpl request_headers{
-      {":method", "POST"},
-      {":path", "/test/long/url"},
-      {":scheme", "http"},
-      {":authority",
-       fmt::format("localhost:{}", fake_upstreams_[0]->localAddress()->ip()->port())}};
-
-  auto response =
-      sendRequestAndWaitForResponse(request_headers, 1024, default_response_headers_, 1024);
-  checkSimpleRequestSuccess(1024, 1024, response.get());
-  testConnectionTiming(response, false, original_usec);
-  EXPECT_EQ(1, test_server_->counter("dns_cache.foo.dns_query_attempt")->value());
-  EXPECT_EQ(1, test_server_->counter("dns_cache.foo.host_added")->value());
-
-  // Now send another request. This should hit the DNS cache.
-  response = sendRequestAndWaitForResponse(request_headers, 512, default_response_headers_, 512);
-  checkSimpleRequestSuccess(512, 512, response.get());
-  testConnectionTiming(response, true, original_usec);
-  EXPECT_EQ(1, test_server_->counter("dns_cache.foo.dns_query_attempt")->value());
-  EXPECT_EQ(1, test_server_->counter("dns_cache.foo.host_added")->value());
-  // Make sure dns timings are tracked for cache-hits.
-  ASSERT_FALSE(response->headers().get(Http::LowerCaseString("dns_start")).empty());
-  ASSERT_FALSE(response->headers().get(Http::LowerCaseString("dns_end")).empty());
+TEST_P(ProxyFilterIntegrationTest, RequestWithBody) { requestWithBodyTest(); }
+
+// Do a sanity check using the getaddrinfo() resolver.
+TEST_P(ProxyFilterIntegrationTest, RequestWithBodyGetAddrInfoResolver) {
```

Comment:
Can we do RequestWithUnknownDomain as well, to have coverage of non-injected lookup failure?

Reply:
Yup will do.